sirepo/package_data/static/js/sirepo.js

```diff
@@ -622,86 +622,77 @@ SIREPO.app.factory('appDataService', function() {
 });
 
-SIREPO.app.factory('notificationService', function($cookies, $sce) {
+SIREPO.app.factory('notificationService', function(cookieService, $sce) {
 
-    var recurrentNotificationTimeout = 1*24*60*60*1000;
-    var nonrecurrentNotificationTimeout = 5*365*24*60*60*1000;
-    var firstVisitCookie = 'net.sirepo.first_visit';
     var self = {};
 
     self.notifications = {};
 
-    var now = new Date();
-    if(! $cookies.get(firstVisitCookie)) {
-        $cookies.put(firstVisitCookie, now.getTime(), {expires: new Date(now.getTime() + nonrecurrentNotificationTimeout)});
-    }
-
     self.addNotification = function(notification) {
         if(! notification.name || ! notification.content) {
             return;
         }
-        if(notification.recurs && ! notification.timeout) {
-            notification.timeout = recurrentNotificationTimeout;
+        self.notifications[notification.name] = notification;
+    };
+
+    self.dismiss = function(name) {
+        if(name) {
+            self.dismissNotification(self.getNotification(name));
         }
-        if(! notification.recurs && ! notification.timeout) {
-            notification.timeout = nonrecurrentNotificationTimeout;
+    };
+
+    self.dismissNotification = function(notification) {
+        if(notification) {
+            self.sleepNotification(notification);
         }
-        self.notifications[notification.name] = notification;
+    };
+
+    self.getContent = function(name) {
+        if(! name || ! self.getNotification(name)) {
+            return '';
+        }
+        return $sce.trustAsHtml(self.getNotification(name).content);
     };
 
     self.getNotification = function(name) {
         return self.notifications[name];
     };
 
     self.removeNotification = function(notification) {
-        $cookies.remove(notification.name);
         delete self.notifications[notification.name];
+        cookieService.removeCookie(cookieDef(notification));
     };
 
     self.shouldPresent = function(name) {
         var now = new Date();
         var notification = self.notifications[name];
-        if(! notification) {
-            return false;
-        }
-        if(! notification.active) {
+        if(! notification || ! notification.active) {
             return false;
         }
 
-        var c = $cookies.get(name);
-        var v = parseInt($cookies.get(firstVisitCookie));
-        if(! c || c === undefined) {
-            return now.getTime() > v + (notification.delay || 0);
+        var cd = cookieDef(notification);
+        var c = cookieService.cleanExpiredCookie(cd);
+        var vcd = SIREPO.APP_SCHEMA.cookies.firstVisit;
+        var vc = cookieService.getCookie(vcd);
+        if(! c) {
+            var lstVisitDays = vc.t - cookieService.timeoutOrDefault(vcd);
+            // we need millisecond comparison here
+            var s = now.getTime() > (lstVisitDays + (notification.delay || 0)) * SIREPO.APP_SCHEMA.constants.oneDayMillis;
+            return s;
```

Comment:
I'd suggest keeping the number of temporary variables to a minimum. Otherwise you don't know easily if variable "c" is used in many places, and it isn't easily searched.
```
if (! cookieService.cleanExpiredCookie(cookieDef(notification))) {
    var vcd = SIREPO.APP_SCHEMA.cookies.firstVisit;
    var vc = cookieService.getCookie(vcd);
    var lstVisitDays = vc.t - cookieService.timeoutOrDefault(vcd);
    // we need millisecond comparison here
    return now.getTime() > (lstVisitDays + (notification.delay || 0)) * SIREPO.APP_SCHEMA.constants.oneDayMillis;
}
```